### DISCUSSION TITLE
Installation possible on macOS?

### DISCUSSION THREAD

**Discussion by jambudipa** [pgpartman/pg_partman#636] [Q&A] [answered]
I am running macOS 13.4, and on running `make install`, I get the error:
```
arm64-apple-darwin20.0.0-clang -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument -Wno-compound-token-split-by-macro -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/marknorgate/anaconda3/include -fdebug-prefix-map=/var/folders/nz/j6p8yfhx1mv_0grj5xl4650h0000gp/T/abs_4737ouxnal/croot/postgresql-split_1685996209501/work=/usr/local/src/conda/postgresql-split-12.15 -fdebug-prefix-map=/Users/marknorgate/anaconda3=/usr/local/src/conda-prefix  -I. -I./ -I/Users/marknorgate/anaconda3/include/server -I/Users/marknorgate/anaconda3/include/internal  -D_FORTIFY_SOURCE=2 -isystem /Users/marknorgate/anaconda3/include -mmacosx-version-min=11.1  -I/Users/marknorgate/anaconda3/include  -c -o src/pg_partman_bgw.o src/pg_partman_bgw.c
make: arm64-apple-darwin20.0.0-clang: No such file or directory
make: *** [src/pg_partman_bgw.o] Error 1
```
Not quire sure why it's looking in the `anaconda3` folder.

I am guessing it needs `arm64-apple-darwin20.0.0-clang`? Having done a `find` for this, it is not on my computer.

I have done `xcode-select --install`, but it seems everything is already installed.

**Comment by jambudipa** [comment]
Here is the output from `clang --version`:
```
clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.5.0
Thread model: posix
```
Can't see for the life of me where this is referenced anywhere in the repo. Have not done C for 30 years.

**Comment by keithf4** [comment]
Not on a Mac myself so consulted with some coworkers. The original compiler/OS version flags postgres was built with would have been recorded in `pg_config` (so part of the postgres install). So it is trying to use the version of the SDK from there, not what is currently installed; you may be able to download the SDK this version of postgres was built with and successfully build at that point.

Where did you originally acquire Postgres?

**Comment by jambudipa** [comment]
Well I had installed it from the downloaded file from the Postgres website; that gave a similar, more elaborate error.

So this message is from the homebrew installation.

I'm going to try with the docker image today. But I can't imagine it's picking that compiler config from my machine, since that is not the available clang?

**Comment by keithf4** [comment]
Apologies, really not sure what's happening here and since I'm not on a Mac it's hard to diagnose. I've labeled this as Help Wanted to see if anyone else that comes around might be able to assist.

If you do find a solution, I would appreciate if you shared it here as well.

**Comment by keithf4** [comment]
Had someone else report some issues on Mac and a solution was found here - https://github.com/pgpartman/pg_partman/issues/596

Not sure if this applies in your case as well

**Comment by djdarkbeat** [comment, marked as answer]
This is how I install pg_partman into Mac OS for any of the different versions I am running in the Mac OS Postgres app as well as Brew outside of the app, I just put the path to the lib for the version I want to install it for.

> PATH="/Applications/Postgres.app/Contents/Versions/14/bin:$PATH" pgxn install pg_partman

You can see that if you run

 > PATH="/Applications/Postgres.app/Contents/Versions/14/bin:$PATH" pg_config

```rb
BINDIR = /Applications/Postgres.app/Contents/Versions/14/bin
DOCDIR = /Applications/Postgres.app/Contents/Versions/14/share/doc/postgresql
HTMLDIR = /Applications/Postgres.app/Contents/Versions/14/share/doc/postgresql
INCLUDEDIR = /Applications/Postgres.app/Contents/Versions/14/include
PKGINCLUDEDIR = /Applications/Postgres.app/Contents/Versions/14/include/postgresql
INCLUDEDIR-SERVER = /Applications/Postgres.app/Contents/Versions/14/include/postgresql/server
LIBDIR = /Applications/Postgres.app/Contents/Versions/14/lib
PKGLIBDIR = /Applications/Postgres.app/Contents/Versions/14/lib/postgresql
LOCALEDIR = /Applications/Postgres.app/Contents/Versions/14/share/locale
MANDIR = /Applications/Postgres.app/Contents/Versions/14/share/man
SHAREDIR = /Applications/Postgres.app/Contents/Versions/14/share/postgresql
SYSCONFDIR = /Applications/Postgres.app/Contents/Versions/14/etc/postgresql
PGXS = /Applications/Postgres.app/Contents/Versions/14/lib/postgresql/pgxs/src/makefiles/pgxs.mk
CONFIGURE =  '--prefix=/Applications/Postgres.app/Contents/Versions/14' '--with-includes=/Applications/Postgres.app/Contents/Versions/14/include' '--with-libraries=/Applications/Postgres.app/Contents/Versions/14/lib' '--enable-thread-safety' '--with-openssl' '--with-bonjour' '--with-libxml' '--with-libxslt' '--with-python' '--with-readline' '--with-uuid=e2fs' '--with-icu' '--with-extra-version= (Postgres.app)' '--with-lz4' 'CFLAGS= -Os -mmacosx-version-min=10.12 -arch arm64 -arch x86_64' 'CXXFLAGS= -Os -mmacosx-version-min=10.12 -arch arm64 -arch x86_64' 'PKG_CONFIG_LIBDIR=/Applications/Postgres.app/Contents/Versions/14/lib/pkgconfig' 'ICU_CFLAGS=-I/Applications/Postgres.app/Contents/Versions/14/share/icu' 'ICU_LIBS=-licui18n -licuuc' 'PYTHON=/Library/Frameworks/Python.framework/Versions/3.9/bin/python3'
CC = gcc
CPPFLAGS = -I/Applications/Postgres.app/Contents/Versions/14/share/icu -I/Applications/Postgres.app/Contents/Versions/14/include/libxml2 -I/Applications/Postgres.app/Contents/Versions/14/include -I/Applications/Postgres.app/Contents/Versions/14/include
CFLAGS = -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Werror=unguarded-availability-new -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -Wno-unused-command-line-argument  -Os -mmacosx-version-min=10.12 -arch arm64 -arch x86_64
CFLAGS_SL =
LDFLAGS = -L/Applications/Postgres.app/Contents/Versions/14/lib -L/Applications/Postgres.app/Contents/Versions/14/lib -L/Applications/Postgres.app/Contents/Versions/14/lib -Wl,-dead_strip_dylibs
LDFLAGS_EX =
LDFLAGS_SL =
LIBS = -lpgcommon -lpgport -llz4 -lxslt -lxml2 -lssl -lcrypto -lz -lreadline -lm
VERSION = PostgreSQL 14.11 (Postgres.app)
```
You get the ability to make sure that everything is correct before running it.  In my case I have Pg 13 - 16 on Mac, and I can run different versions of extensions in each version of Postgres.  Since I don't need this PATH configured all the time I just use It in the CLI for installing stuff. @keithf4 @jambudipa

**Comment by keithf4** [comment]
Thanks very much for the suggestion! I'm going to move this into a discussion Q &A for future reference
